### PR TITLE
TDD - Mandatory Attendees

### PR DESCRIPTION
**Pull request by 0tpyrk**
Completed first version of meeting scheduling algorithm and wrote tests for optional attendees.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 1]
```diff
@@ -270,5 +271,112 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  /*@Test
```
Comment (Yunni) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:275]: Why comment this out?

Comment (0tpyrk) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:275]: These are the new tests for the second half of the project including optional attendees, which my code hasn't been programmed to accommodate yet.

Comment (tc466) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:275]: Maybe move to a separate PR?

Comment (0tpyrk) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:275]: Okay!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,84 @@
...
+        // if equal starts, compare ending times
+        } else if (TimeRange.ORDER_BY_START.compare(e.getWhen(), personRelevantEvents.get(i).getWhen()) == 0) {
+          if (TimeRange.ORDER_BY_END.compare(e.getWhen(), personRelevantEvents.get(i).getWhen()) > 0) {
+            e = personRelevantEvents.get(i);
+          }
+        }
+      }
+      sortedRelevantEvents.add(e);
+      personRelevantEvents.remove(e);
+    }
```
[the remark this answers is not in the document]
Comment (0tpyrk) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:58]: Good idea, thanks!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -15,9 +15,84 @@
...
+
+    // test if sort works
+    System.out.println("Beginning");
+    for (Event e : sortedRelevantEvents) {
+        System.out.println(e.getWhen().start());
+    }
+
+    int start = TimeRange.START_OF_DAY;
+    int end;
```
[the remark this answers is not in the document]
Comment (0tpyrk) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:78]: Got it!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -15,9 +15,84 @@
...
+    }
+    for (Event e : toDelete) {
+      sortedRelevantEvents.remove(e);
+    }
+
+    // test if sort works
+    System.out.println("Beginning");
+    for (Event e : sortedRelevantEvents) {
+        System.out.println(e.getWhen().start());
+    }
```
[the remark this answers is not in the document]
Comment (0tpyrk) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:75]: Got it!

**Pull request merged**